bleak/backends/corebluetooth/client.py

```diff
@@ -80,12 +79,17 @@ async def connect(self, **kwargs) -> bool:
                 raise BleakError(
                     "Device with address {} was not found".format(self.address)
                 )
-
-        logger.debug("Connecting to BLE device @ {}".format(self.address))
-
-        manager = self._device_info.manager().delegate()
-        await manager.connect_(self._device_info)
-        manager.disconnected_callback = self._disconnected_callback_client
+        # self._device_info.manager() should return a CBCentralManager
+        if hasattr(self._device_info.manager(), "delegate"):
```

Comment:
In https://github.com/hbldh/bleak/issues/319#issuecomment-704403926 it is mention that `self._device_info.manager()` is `None` rather than just missing the `delegate` attribute. So maybe check for `is not None` instead?

Also, I'm wondering if adding this sort of check is just hiding a deeper issue.

Reply:
> Also, I'm wondering if adding this sort of check is just hiding a deeper issue.


It raises a `BleakError` with the same message as if the device was not detected. Maybe it would be worth to change it.

Also I though of another check when a `CBPeripheral` object is discovered to assert that calling `.manager()` should return a `CBCentralManager` instance instead of `None`?

I agree with the deeper issue. But it is hard to debug without the device that is failing. Or how would you reproduce this?
It may be CoreBluetooth itself that fails silently? Maybe in console logging appear some sort of error message?

Reply:
Looking at https://developer.apple.com/documentation/corebluetooth/cbperipheral?language=objc, I don't see that `manager` is a property or method on `CBPeriperal`. Could we be using an undocumented feature? Maybe we need to keep track of the `CBCentralManager` and the `CBPeriperal` objects separately? Or am I missing something?

Reply:
>  I don't see that manager is a property or method on `CBPeriperal`.  Could we be using an undocumented feature? 

Yes that seemed strange to me too. 🙄 

> Maybe we need to keep track of the CBCentralManager and the CBPeriperal objects separately? Or am I missing something?

This could be a solution, I've just done a test passing directly `CentralManagerDelegate` to BLEDevice and it seems to work nice.

Updating this PR now with these changes 👍